internal/uvm/scsi.go

```diff
@@ -261,9 +293,12 @@ func (uvm *UtilityVM) AddSCSIPhysicalDisk(ctx context.Context, hostPath, uvmPath
 //
 // `vmAccess` indicates what access to grant the vm for the hostpath
 //
+// `evdType` indicates the `ExtensibleVirtualDiskType` if `attachmentType` is "ExtensibleVirtualDisk".
+// it should be empty otherwise.
+//
 // Returns result from calling modify with the given scsi mount
-func (uvm *UtilityVM) addSCSIActual(ctx context.Context, hostPath, uvmPath, attachmentType string, readOnly bool, guestOptions []string, vmAccess VMAccessType) (sm *SCSIMount, err error) {
-	sm, existed, err := uvm.allocateSCSIMount(ctx, readOnly, hostPath, uvmPath, attachmentType, vmAccess)
+func (uvm *UtilityVM) addSCSIActual(ctx context.Context, hostPath, uvmPath, attachmentType string, readOnly bool, guestOptions []string, vmAccess VMAccessType, evdType string) (sm *SCSIMount, err error) {
```

Comment:
I don't really like how many parameters this function has, I feel like at this point, we should have an internal scsi options struct or something? 8 parameters is a lot

Reply:
Fixed.